pkg/fileutils/fileutils.go

```diff
@@ -1,26 +1,56 @@
 package fileutils
 
 import (
-	"github.com/Sirupsen/logrus"
 	"path/filepath"
+
+	"github.com/Sirupsen/logrus"
 )
 
 // Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
 func Matches(relFilePath string, patterns []string) (bool, error) {
-	for _, exclude := range patterns {
-		matched, err := filepath.Match(exclude, relFilePath)
+
+	matched := false
+
+	for _, pattern := range patterns {
+
+		var negative bool
+		if pattern == "" {
+			continue
+		}
+
+		if pattern[0] == '!' {
+			if len(pattern) == 1 {
+				continue
+			}
+			negative = true
+			pattern = pattern[1:]
+		}
```

Comment:
This is a tough one.  A warning might be nice but within this func I don't think we access to the output stream. So, if I had to choose between silently ignoring it and stopping/erroring, I'd pick error and halt the build. I think skipping what is really an error case w/o telling the user is too close to lying for my tastes :-)